src/persist-client/src/iter.rs

```diff
@@ -449,7 +450,9 @@ impl<T: Timestamp + Codec64 + Lattice, D: Codec64 + Semigroup> Consolidator<T, D
                         }
                         self.metrics.consolidation.parts_fetched.inc();
                         *part = ConsolidationPart::from_encoded(
-                            handle.await??,
+                            handle
+                                .await?
+                                .ok_or_else(anyhow!("unexpectedly missing part"))?,
```

Comment:
I think I'm mildly inclined away from this second commit ("Hoist up error handling on a missing part") because I think it's quite useful to have the key in the error message and the before seems more likely to ensure that happens. Otoh, I feel like it's likely that I'm missing context here (presumably the skunkwork), so curious what your motivation was!

That said, I certainly would not block merging over this, as long as we can get the key back in all the new error message paths

Reply:
> curious what your motivation was!

In brief: right now the only error that gets raised by fetching is "blob did not exist", but that's not reflected in the very broad `anyhow::Result` type. Narrowing the error result type is intended to make it safer to handle downstream in ways other than panicking. (Which was the skunkwork!)

> as long as we can get the key back in all the new error message paths

I reworked the branch to return the `BlobKey` that wasn't found. This does make it easier to report the full missing key, while also hopefully keeping the range of possible errors clearer.

Reply:
(If that still doesn't feel good, that's cool -- I'll pick this up again later when I have a more specific usage to point to.)

Reply:
Nope, makes sense and the new version addresses my concern nicely!